### PR TITLE
Address install inconsistencies between documentation and pip requirements

### LINKED ISSUE #8058: Fix consistency between install docs and package requirements list
I'm sure we had issues about this in the past, but the docs and the now cleaned up `pip-requirements` files got out of sync again as it got discovered in https://github.com/astropy/astropy/pull/8048

There still seems to be a bit of discrepancies between the install docs and the requirement files, we'll need to fix these before the release:

In `install.rst` but not in any of the pip-requirements files:

 -   `xmllint`
 -   `soretedcontainers`
 -   `setuptools`
 -   `objgraph`

Listed in one of the pip-requirements but not in install.rst.

  -  `coverage`
  -  `skyfield`
  -  `pytest-xdist`
  -  `pytest-mpl`
  -  `ipython`
  -  `pillow`

Some of these are clearly there for testing purposes, so we probably don't need to list them, but then we don't need to list e.g. setuptools either.

### PR DESCRIPTION
**Pull request by pllim**
Fix #8058

### CODE REVIEW

File: pip-requirements-doc [hunk 1]
```diff
@@ -7,7 +7,6 @@ scikit-image
 pandas
 pytz
 beautifulsoup4
-ipython
```
Comment (pllim) [inline pip-requirements-doc]: @saimn , why did you include `ipython` in #7711 ? I don't see why it is needed for doc build. Do you remember?

Comment (saimn) [reply pip-requirements-doc]: I don't know if it is needed, but it was in the conda environment (https://github.com/astropy/astropy/pull/7711/files#diff-f6547eedc6811e1dd247024c3940b0c3L24).

Comment (bsipocz) [reply pip-requirements-doc]: we definitely need it for the tests (https://github.com/astropy/astropy/blob/51c619e5ffe7aaf748e7c34d08c6336762decc74/astropy/table/tests/test_jsviewer.py#L159)

**Comment by pllim** [comment]
Thanks, @saimn and @bsipocz . I have added IPython back in and included a note in doc.

File: docs/install.rst [hunk 2]
```diff
@@ -73,6 +74,24 @@ Astropy also depends on other packages for optional features:
...
+- `pytest-astropy <https://github.com/astropy/pytest-astropy>`_: See
+  :ref:`sourcebuildtest`.
+
+- `pytest-xdist <https://docs.pytest.org/en/3.0.0/xdist.html>`_: Used for
+  distributed testing.
+
+- `pytest-mpl <https://github.com/matplotlib/pytest-mpl>`_: Used for testing
+  with Matplotlib figures.
+
+- `IPython <https://ipython.org/>`_: Used for testing notebook interface of
```
Comment (bsipocz) [inline docs/install.rst]: "duplicate target name" warning, add another underscore at the end to make the warning go away.

Comment (pllim) [reply docs/install.rst]: Ops, thanks!

**Comment by bsipocz** [comment]
Thanks @pllim!

**Pull request merged**